### ISSUE TITLE
Incorrectly parsing extended tweet from stream

### ISSUE THREAD

**Issue by kokarn** [pirxpilot/tweet-html#1]
Hey!
Great library you've got here. I'm using it to parse tweets from the Stream api in node.

Something that'd be nice to look into tho is tweets that contain the extended data under `extended_tweet`. I'll and an example at the end that's directly from the Stream API.

Should be pretty straightforward actually. Just check for `extended_tweet` and then use `extended_tweet.full_text` instead of `text` and `extended_tweet.entities` instead of `entities`.

```json
{
    "created_at": "Tue Feb 06 15:25:48 +0000 2018",
    "id": 960897321608339500,
    "id_str": "960897321608339456",
    "text": "@HerrLeGnO If you suspect a player of cheating please report them in game and to our support page if you have addit… https://t.co/11hY9mETK2",
    "display_text_range": [
        11,
        140
    ],
    "source": "<a href=\"http://www.zendesk.com\" rel=\"nofollow\">Zendesk</a>",
    "truncated": true,
    "in_reply_to_status_id": 959857680872591400,
    "in_reply_to_status_id_str": "959857680872591362",
    "in_reply_to_user_id": 948192552,
    "in_reply_to_user_id_str": "948192552",
    "in_reply_to_screen_name": "HerrLeGnO",
    "user": {
        "id": 882248998547071000,
        "id_str": "882248998547070981",
        "name": "PUBG Help",
        "screen_name": "PUBG_help",
        "location": "Erangel",
        "url": "http://forums.playbattlegrounds.com",
        "description": "Official support account for @PUBATTLEGROUNDS",
        "translator_type": "none",
        "protected": false,
        "verified": false,
        "followers_count": 30414,
        "friends_count": 30,
        "listed_count": 53,
        "favourites_count": 832,
        "statuses_count": 7305,
        "created_at": "Tue Jul 04 14:45:27 +0000 2017",
        "utc_offset": 0,
        "time_zone": "London",
        "geo_enabled": false,
        "lang": "en-gb",
        "contributors_enabled": false,
        "is_translator": false,
        "profile_background_color": "000000",
        "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
        "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
        "profile_background_tile": false,
        "profile_link_color": "333333",
        "profile_sidebar_border_color": "000000",
        "profile_sidebar_fill_color": "000000",
        "profile_text_color": "000000",
        "profile_use_background_image": false,
        "profile_image_url": "http://pbs.twimg.com/profile_images/896058452870664192/HFvbJD0v_normal.jpg",
        "profile_image_url_https": "https://pbs.twimg.com/profile_images/896058452870664192/HFvbJD0v_normal.jpg",
        "profile_banner_url": "https://pbs.twimg.com/profile_banners/882248998547070981/1511326896",
        "default_profile": false,
        "default_profile_image": false,
        "following": null,
        "follow_request_sent": null,
        "notifications": null
    },
    "geo": null,
    "coordinates": null,
    "place": null,
    "contributors": null,
    "is_quote_status": false,
    "extended_tweet": {
        "full_text": "@HerrLeGnO If you suspect a player of cheating please report them in game and to our support page if you have additional evidence such as video. -D\n\nhttps://t.co/6BP0XzySCn",
        "display_text_range": [
            11,
            172
        ],
        "entities": {
            "hashtags": [],
            "urls": [
                {
                    "url": "https://t.co/6BP0XzySCn",
                    "expanded_url": "https://pubgsupport.zendesk.com/hc/en-us",
                    "display_url": "pubgsupport.zendesk.com/hc/en-us",
                    "indices": [
                        149,
                        172
                    ]
                }
            ],
            "user_mentions": [
                {
                    "screen_name": "HerrLeGnO",
                    "name": "Alejandro LeGnØ",
                    "id": 948192552,
                    "id_str": "948192552",
                    "indices": [
                        0,
                        10
                    ]
                }
            ],
            "symbols": []
        }
    },
    "quote_count": 0,
    "reply_count": 0,
    "retweet_count": 0,
    "favorite_count": 0,
    "entities": {
        "hashtags": [],
        "urls": [
            {
                "url": "https://t.co/11hY9mETK2",
                "expanded_url": "https://twitter.com/i/web/status/960897321608339456",
                "display_url": "twitter.com/i/web/status/9…",
                "indices": [
                    117,
                    140
                ]
            }
        ],
        "user_mentions": [
            {
                "screen_name": "HerrLeGnO",
                "name": "Alejandro LeGnØ",
                "id": 948192552,
                "id_str": "948192552",
                "indices": [
                    0,
                    10
                ]
            }
        ],
        "symbols": [],
        "media": []
    },
    "favorited": false,
    "retweeted": false,
    "possibly_sensitive": false,
    "filter_level": "low",
    "lang": "en",
    "timestamp_ms": "1517930748737"
}
```

**Comment by pirxpilot** [comment]
Thanks!

I am fairly sure that the library parses correctly streams requested with `tweet_mode=extended` parameter. At least according to the [Twitter docs][1] Could you try that and let ma know if you stil have an issue.

That said - I'll consider PR supporting other tweet formats. And I'll add `tweet_mode` info to Readme.

[1]: https://developer.twitter.com/en/docs/tweets/tweet-updates

**Comment by kokarn** [comment]
It does if they are requested over the regular REST API.

However, the "Compatibility with additional extended_tweet in payload" is used for the [Streaming API](https://developer.twitter.com/en/docs/tweets/filter-realtime/overview) and produces the above example payload which is not parsed correctly, as it doesn't contain the default `extended_entities` on the root object.

Also, the standard `extended` mode isn't available for the Streaming APIs.

I might look into a PR, but no promises :)

**Comment by pirxpilot** [comment]
Ahh - makes sense.

I might look into it, but no promises :):)

**Comment by pirxpilot** [comment]
Fixed by #4
